pkg/psalabelsyncer/privileged_namespaces_controller.go

```diff
@@ -0,0 +1,72 @@
+package psalabelsyncer
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	corev1apply "k8s.io/client-go/applyconfigurations/core/v1"
+	corev1informers "k8s.io/client-go/informers/core/v1"
+	corev1client "k8s.io/client-go/kubernetes/typed/core/v1"
+	corev1listers "k8s.io/client-go/listers/core/v1"
+	psapi "k8s.io/pod-security-admission/api"
+
+	"github.com/openshift/library-go/pkg/controller/factory"
+	"github.com/openshift/library-go/pkg/operator/events"
+)
+
+const privilegedControllerName = "privileged-namespaces-psa-label-syncer"
+
+type privilegedNamespacesPSALabelSyncer struct {
+	nsClient corev1client.NamespaceInterface
+	nsLister corev1listers.NamespaceLister
+}
+
+func NewPrivilegedNamespacesPSALabelSyncer(
+	namespaceClient corev1client.NamespaceInterface,
+	namespaceInformer corev1informers.NamespaceInformer,
+	eventRecorder events.Recorder,
+) factory.Controller {
+	c := &privilegedNamespacesPSALabelSyncer{
+		nsClient: namespaceClient,
+		nsLister: namespaceInformer.Lister(),
+	}
+
+	return factory.New().
```

Comment:
you sure you want to layer on this?  I won't nak it, but if it's only saving 50 lines of boiler plate are you sure it's worth the trust that it works?